readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/offset/HeartbeatConsumer.java

```diff
@@ -68,6 +57,21 @@ public void start(Thread.UncaughtExceptionHandler uncaughtExceptionHandler, Stri
         runningThread.start();
     }
 
+    public void run() {
+        synchronized (latestPartitionsOffset) {
```

Comment:
we should not synchronise latestPartitionOffset anymore. It will always be called by the consumer thread (if we remove Heartbeat consumer thread of course)

Reply:
done